File: ios/RNLiveChat.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNLiveChat, NSObject)

RCT_EXTERN_METHOD(initialize:(NSString*)licenseId)

RCT_EXTERN_METHOD(setGroup:(NSString*)groupId)

RCT_EXTERN_METHOD(presentChat)

RCT_EXTERN_METHOD(setCustomer:(NSString*)name email:(NSString*)email)

RCT_EXTERN_METHOD(setVariable:(NSString*)key value:(NSString*)value)

@end

File: ios/RNLiveChat.swift
import LiveChat

@objc(RNLiveChat)
class RNLiveChat: NSObject {

    @objc(initialize:)
    func initialize(licenseId: String) -> Void {
        DispatchQueue.main.async {
            LiveChat.licenseId = licenseId
        }
    }
    
    @objc(setGroup:)
    func setGroup(groupId: String) -> Void {
        DispatchQueue.main.sync {
            LiveChat.groupId = groupId
        }
    }
    
    @objc(presentChat)
    func presentChat() -> Void {
        DispatchQueue.main.async {
            LiveChat.presentChat()
        }
    }
    
    @objc(setCustomer:email:)
    func setCustomer(name: String, email: String) -> Void {
        DispatchQueue.main.async {
            LiveChat.name = name
            LiveChat.email = email
        }
    }
    
    @objc(setVariable:value:)
    func setVariable(key: String, value: String) -> Void {
        DispatchQueue.main.async {
            LiveChat.setVariable(withKey: key, value: value)
        }
    }
}
